File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left = 0
        right = x

        while left < right:
            mid = (right + left)//2
            compute = mid * mid
            if compute == x:
                return mid
            if compute < x:
                left = mid + 1
            else:
                right = mid - 1

        return right if right * right <= x else right - 1



        

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {

        int left = 0;
        int right = 1 + (x/2);

        while(left<=right){

            int mid = left + (right-left)/2;
            double square = (double)mid*mid;

            if(square==x){
                return mid;
            }

            if(square<x){
                left = mid+1;
            }else{
                right = mid - 1;
            }

        }

        return right;
        
    }
};